packagedb/api.py

```diff
@@ -644,3 +673,87 @@ def _get_enhanced_package(package, packages):
 class PackageSetViewSet(viewsets.ReadOnlyModelViewSet):
     queryset = PackageSet.objects.prefetch_related('packages')
     serializer_class = PackageSetAPISerializer
+
+def get_resolved_purls(purls):
+    unique_resolved_purls = set()
+    unsupported_purls = set()
+    unsupported_vers = set()
+
+    for items in purls or []:
+        purl = items.get('purl')
+        vers = items.get('vers')
+        
+        try:
+            parsed_purl = PackageURL.from_string(purl)
+        except ValueError:
+            unsupported_purls.add(purl)
+            continue
+
+        if parsed_purl.version:
+            unique_resolved_purls.add(purl)
+            continue
+
+        if resolved:= resolve_verse(parsed_purl, vers):
+            unique_resolved_purls.update(resolved)
+        else:
+            unsupported_vers.add(vers)
+
+    return list(unique_resolved_purls), list(unsupported_purls), list(unsupported_vers)
+
+
+
+def resolve_verse(parsed_purl, vers):
+    try:
+        version_range = VersionRange.from_string(vers)
+    except ValueError:
+        return
+
+    if not version_range.constraints:
+        return
+
+    all_versions = get_all_versions(parsed_purl) or []
+
+    return [
+        str(
+            PackageURL(
+                type=parsed_purl.type,
+                namespace=parsed_purl.namespace,
+                name=parsed_purl.name,
+                version=version.string,
+            )
+        )
+        for version in all_versions
+        if version in version_range
+    ]
+
+def get_all_versions(purl: PackageURL):
+    if (
+        purl.type not in VERSION_API_CLASSES_BY_PACKAGE_TYPE
+        or purl.type not in VERSION_CLASS_BY_PACKAGE_TYPE
+    ):
+        return
+
+    package_name = get_api_package_name(purl)
+    versionAPI = get_version_fetcher(purl)
+    
+    if not package_name or not versionAPI:
+        return    
+
+    all_versions = versionAPI().fetch(package_name)
+    versionClass = VERSION_CLASS_BY_PACKAGE_TYPE.get(purl.type)
+
+    return [versionClass(package_version.value) for package_version in all_versions]
+
+
+VERSION_CLASS_BY_PACKAGE_TYPE = {
```

Comment:
Can we get this from univers instead?
 Either patch univers to expose a VERSION_CLASS_BY_PACKAGE_TYPE or provide a function there that returns a version class for a type. This should in all cases be computed and not hardcoded.

If you use the built-in mapping and use the version class attribute you will get what you need
https://github.com/nexB/univers/blob/205da7ecbf7b0f195662373ea710b2b84a877eb0/src/univers/version_range.py#L417 
```suggestion
VERSION_CLASS_BY_PACKAGE_TYPE = {pkg_type: range_class.version_class for pkg_type, range_class in RANGE_CLASS_BY_SCHEMES.items()}
```

```